core/tokenrate/bancor.go

```diff
@@ -69,27 +69,77 @@ func (qq *bancorQuoteQuery) getUSD(ctx context.Context, symbol string) (float64,
 	rate, ok := result.Data.Rate24hAgo["usd"]
 
 	if ok {
-		if rate == 0 {
+
+		if rate.Value == 0 {
 			rate, ok = result.Data.Rate["usd"]
 			if ok {
```

Comment:
Seems should be `if !ok`

Reply:
No. It is ‘ok’.

Reply:
refactored code.it should be more clean now.